### PR TITLE
Adds Backstage integration documentation

### PR DESCRIPTION
**Pull request by jorgelainfiesta**
### What does this PR do?

The Backstage integration allows you to embed dashboards and graphs from DataDog into your Backstage instance.

### Motivation

Backstage, an open platform for building Developer Platforms, is growing rapidly after reaching 1.0 and being accepted as a CNCF Incubation project. DataDog is commonly used by Backstage users, and this integration makes it easy to bring graphs into Backstage instances.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional notes

This integration relies on public URLs from DataDogs dashboard. The integration [implementation is on Backstage's side](https://github.com/RoadieHQ/roadie-backstage-plugins/tree/main/plugins/frontend/backstage-plugin-datadog) and doesn't use any DataDog API.

### CODE REVIEW

**Comment by buraizu** [comment]
Thanks for the PR, created DOCS-3600 to begin the editorial review process

File: backstage/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# CHANGELOG - Backstage Integration
+
+2.0.0
```
Comment (sarah-witt) [inline backstage/CHANGELOG.md:3]: Nit- why does this start at version 2.0.0?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -13,11 +13,12 @@
 /amixr/                                  @iskhakov ildar@amixr.io
 /apache-apisix/                          @bisakhmondal bisakhmondal00@gmail.com
 /apollo/                                 @sachindshinde sachin@apollographql.com
-/appkeeper/                              @numno1
+/appkeeper/                              @numno1 support@roadie.io
```
Comment (sarah-witt) [inline .github/CODEOWNERS:16]: Why are the appkeeper codeowners updated here?

**Review by sarah-witt — COMMENTED** [review]
Looks great, I just had two small comments!

**Comment by jorgelainfiesta** [comment]
Thanks for the review @sarah-witt! I've updated the PR

File: .github/CODEOWNERS (hunk 2 above)
Comment (sarah-witt) [reply .github/CODEOWNERS]: No problem, thanks!

File: backstage/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  name: sample-service
+  description: |
+    A sample service
+  annotations:
+    datadoghq.com/dashboard-url: <<DATADOGURL>>
+```
+
+### Screenshot
+
+![Screenshot of Backstage's Overview featuring a DataDog graph](https://raw.githubusercontent.com/RoadieHQ/roadie-backstage-plugins/main/plugins/frontend/backstage-plugin-datadog/docs/datadog-widget.png)
```
Comment (sarah-witt) [inline backstage/README.md:35]: Can you move this image into our repo so it can be displayed correctly? Thanks!

**Review by sarah-witt — COMMENTED** [review]
Thanks! I had one more comment

File: backstage/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (buraizu) [reply backstage/README.md:35]: Would you be able to provide a more descriptive title and additional detail to this section? Currently it's not clear what information is meant to be conveyed by the screenshot. 

For example, would a customer see something similar to the screenshot after successfully setting this up? If so, please update the title of this section to `Validation` or something similar, as in this other example from the `integrations-extras` repository:
https://github.com/DataDog/integrations-extras/tree/master/kernelcare#validation

With that being said, we generally caution against including screenshots in documentation, as these are difficult to maintain due to UI changes. Please consider if the content of this screenshot can be adequately expressed in a written form, and substitute the written description for the screenshot if so.

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for the PR! Made some suggestions to better align the page with our contributing guidelines:
https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#style-guide-for-documentation-site

File: backstage/README.md (hunk 3 above)
Comment (jorgelainfiesta) [reply backstage/README.md:35]: I'll remove the image as suggested by @buraizu, thanks!

**Comment by jorgelainfiesta** [comment]
Thanks for your review @buraizu, I've removed the screenshot as you suggested and replaced it with a text-based validation section.

File: backstage/README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+kind: Component
+metadata:
+  name: sample-service
+  description: |
+    A sample service
+  annotations:
+    datadoghq.com/dashboard-url: <DATADOGURL>
+```
+
+### Verifiaction
```
Comment (buraizu) [inline backstage/README.md:33]:
```suggestion
### Verification
```

File: backstage/README.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+Open your Backstage instance Overview tab and see if your Datadog dahsboard or graph is rendered as expected.
```
Comment (buraizu) [inline backstage/README.md:35]:
```suggestion
Open your Backstage instance Overview tab to confirm your Datadog dashboard or graph is rendered as expected.
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for implementing those suggestions! Just noticed a couple of minor typos and made one more small suggestion.

**Comment by jorgelainfiesta** [comment]
Sorry, that's a bit embarrassing. Thanks!!

**Review by buraizu — APPROVED** [review]
Thanks again! Approved, with one minor edit in the manifest.json

**Comment by steveny91** [comment]
Hello @jorgelainfiesta. I'm going to try deploying this on staging and then checking to see if looks as expected. If everything looks good, is this ready to be deployed? Or is there a timeline you're looking for?

**Comment by jorgelainfiesta** [comment]
Hi @steveny91, this is ready to be deployed, thanks!

File: backstage/CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# CHANGELOG - Backstage Integration
+
+1.0.0
+==================
+### Changes
+
+* [FEATURE] Backstage DataDog Integration
```
Comment (fanny-jiang) [inline backstage/CHANGELOG.md:7]: small nit

```suggestion
* [FEATURE] Backstage Datadog Integration
```

**Review by fanny-jiang — CHANGES_REQUESTED** [review]

File: backstage/CHANGELOG.md (hunk 6 above)
Comment (fanny-jiang) [reply backstage/CHANGELOG.md:7]: Besides this small change, the tile looks good on staging!

**Comment by jorgelainfiesta** [comment]
Thanks @fanny-jiang !

**Review by fanny-jiang — APPROVED** [review]

**Pull request merged**